src/OpenTelemetry/BatchExportProcessorOptions.cs

```diff
@@ -14,11 +14,57 @@
 // limitations under the License.
 // </copyright>
 
+using System;
+using System.Security;
+using OpenTelemetry.Internal;
+
 namespace OpenTelemetry
 {
     public class BatchExportProcessorOptions<T>
         where T : class
     {
+        internal const string MaxQueueSizeEnvVarKey = "OTEL_BSP_MAX_QUEUE_SIZE";
```

Comment:
These are specific to batch span processors, we need some logic to make sure it is not affecting other scenario (e.g. metrics).

Reply:
Do you have any proposal on how to approach this?

I think that the problem is that the root problem is that the .NET SDK tries to use common abstractions to implement multiple OTel specification components. E.g. we have a common exporter base class for both metrics and traces exporter. I am aware it reduces some code duplication but it tightly couples the API.

Can we use `BatchExportProcessorOptions` for tracing (to not break SemVer) and create a `BatchMetricsExportProcessorOptions` for metrics SDK?

Reply:
What if we did something like...

```csharp
        internal const string SpanMaxQueueSizeEnvVarKey = "OTEL_BSP_MAX_QUEUE_SIZE";

        internal const string SpanMaxExportBatchSizeEnvVarKey = "OTEL_BSP_MAX_EXPORT_BATCH_SIZE";

        internal const string SpanExporterTimeoutEnvVarKey = "OTEL_BSP_EXPORT_TIMEOUT";

        internal const string SpanScheduledDelayEnvVarKey = "OTEL_BSP_SCHEDULE_DELAY";

        public BatchExportProcessorOptions()
        {
            if (typeof(T) == typeof(Activity))
            {
               LoadSpanEnvVars();
            }
        }

        private void LoadSpanEnvVars()
        {
            int value;

            if (TryLoadEnvVarInt(SpanExporterTimeoutEnvVarKey, out value))
            {
                this.ExporterTimeoutMilliseconds = value;
            }

            if (TryLoadEnvVarInt(SpanMaxExportBatchSizeEnvVarKey, out value))
            {
                this.MaxExportBatchSize = value;
            }

            if (TryLoadEnvVarInt(SpanMaxQueueSizeEnvVarKey, out value))
            {
                this.MaxQueueSize = value;
            }

            if (TryLoadEnvVarInt(SpanScheduledDelayEnvVarKey, out value))
            {
                this.ScheduledDelayMilliseconds = value;
            }
        }
```

I'm not really a fan of duck typing in generics like this but trying to introduce classes into the hierarchy now will probably be a breaking change.

Reply:
@CodeBlanch thanks 👍 

> but trying to introduce classes into the hierarchy now will probably be a breaking change

That is why I suggest changing the metrics SDK as AFAIK it od not stable yet. And leave the tracing SDK as it is.

Your proposal would work but maybe it is worth to make a step back

Reply:
@CodeBlanch 
I won't be that simple 😞 
 `OtlpMetricsExporter` uses `OtlpExporterOptions` which has a `BatchExportProcessorOptions<Activity>`.

Reply:
After the SIG meeting, I have an impression that this would be the preferred way forward:
1. Create an issue that `OtlpMetricsExporter` should not reference `BatchExportProcessorOptions<Activity>`
1. Create a `BatchSpanExportProcessorOptions` as a subclass of `BatchExportProcessorOptions<Activity>`